.github/workflows/build.yml

```
@@ -0,0 +1,21 @@
+name: Build and Test
+
+on: [push, pull_request]
```

I would prefer if Docker images were made only on a workflow dispatch.

```suggestion
on: 
    workflow_dispatch:
```